### ISSUE TITLE
Grails 3.0.2:  grails run-app -https returns "Unable to automatically generate a SSL certificate, manual configuration required. Set 'server.ssl.key-store' in application.yml"

### ISSUE THREAD

**Issue by ikakara** [grails/grails-core#9018]
Related to this: https://github.com/grails/grails-core/issues/648 
The CNFE has been fixed, but not able to start grails w/ -https

Environment:
--------------------------
WIndows 8.1
Oracle JDK
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)
```
> grails create-app test302
> cd test302
> grails compile
> grails run-app -https
```
Results in:
| Generating SSL certificate
| Warning Unable to automatically generate SSL certificate, manual configuration required. Set 'server.ssl.key-store' in application.yml

After the warning, the console displays: "Grails application running at http://localhost:8443 in environment: development" ... note "http" and port "8443" are now out of synch: http -> 8080 (on dev) and https -> 8443

**Comment by graemerocher** [comment]
The reason is that Grails cannot find the the KeyTool class in your JDK:

https://github.com/grails/grails-profile-repository/blob/773f78f00fcfe1a5c6ee0cd76b40134c5b3e9240/profiles/base/commands/run-app.groovy#L111

The most likely reason is that you are running Grails with a JRE instead of a JDK or that you have an incompatible JDK. In this case as the error message suggests you have to manually configure your SSL certificate for this to work. See here for configuration information:

http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#howto-configure-ssl

**Comment by ikakara** [comment]
Graeme,

I'm able to generate SSL key in Grails 2.5.0 w/ the same (Oracle JDK).  I'm guessing there's a "fix" for this issue for Windows users w/ Oracle JDK 1.8u45 for Grails 3.  

I've tested the followign on Grails 3:
Oracle JDK 1.7.0_79 - it generates the SSL cert fine
Oracle JDK 1.8.0_25 (the earliest I have) - and it gives the same error

So I'm guessing something changed between 1.7 and 1.8.  Maybe they repackaged the KeyTool class ... or don't include the KeyTool class.

As I mentioned, Grails 2.5.0 and JDK 1.8u45 generates the SSL certificate.  So I've been trying to compare the difference in the script RunAppHttps.groovy between Grails 2.5 and Grails 3.0 ... and I haven't been able to locate where KeyTool is being used in 2.5 :(  

You pointed me to the Grails 3 code: https://github.com/grails/grails-profile-repository/blob/773f78f00fcfe1a5c6ee0cd76b40134c5b3e9240/profiles/base/commands/run-app.groovy#L111  

Can you point me to the equivalent code in Grails 2.5.0 ?  I found this https://github.com/grails/grails-core/blob/2.5.x/grails-project-api/src/main/groovy/org/codehaus/groovy/grails/project/container/GrailsProjectRunner.groovy

But I don't understand where the KeyTool is being called.  I've greped the whole project searching for "KeyTool" and I can't locate it.  Can you shed any light?

Allen

**Comment by graemerocher** [comment]
See here https://github.com/grails-plugins/grails-tomcat-plugin/blob/master/src/groovy/org/grails/plugins/tomcat/TomcatServer.groovy#L186

**Comment by ikakara** [comment]
Thanks Graeme!  I merged the 2.5.x code into 3.x - https://github.com/grails/grails-profile-repository/pull/14
